src/codeu/chat/client/simplegui2/ChatMenu.java

```diff
@@ -0,0 +1,200 @@
+                    alertNoUser.display();
+                }
+            }
+        });
+
+        addUser.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+                NewUser user1=new NewUser(clientContext);
+            }
+        });
+
+        newConvo.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+
+                if (clientContext.user.hasCurrent()) {
+
+                    NewConversation addConversationBox = new NewConversation(clientContext, "Add Conversation", "Enter new conversation name:" );
+                    addConversationBox.enterNewConversation();
+                    String s = addConversationBox.conversationInput();
+
+                    if (s != null && s.length() > 0) {
+                        clientContext.conversation.startConversation(s, clientContext.user.getCurrent().id);
+                        //ConversationPanel.this.getAllConversations(listModel);
+                    }
+                } else {
+                    AlertBox alertNoUser = new AlertBox("You are not signed in.", "Please sign in before joining a conversation.");
+                    alertNoUser.display();
+                }
+            }
+        });
+
+
+        //add buttons to menu
+        topbar.getChildren().addAll(signOut, addUser, newConvo);
+
+        return topbar;
+    }
+
+    private static VBox sideBar(){
+
+        //create sidebar
+        VBox sidebar= new VBox(5);
+
+        sidebar.setPadding(new Insets(3));
+        sidebar.setAlignment(Pos.TOP_LEFT);
+        sidebar.setMinWidth(200);
+
+        //Label for box
+        Label sideLabel=new Label("Conversations");
+
+        //border for box
+        sidebar.setStyle("-fx-padding: 10;" +
+                "-fx-border-style: solid inside;" +
+                "-fx-border-width: 4;"  +
+                        "-fx-border-insets: 0;"+
+                "-fx-border-radius: 5;" +
+                "-fx-border-color: #336699;");
+
+        sidebar.getChildren().addAll(sideLabel);
+
+        return sidebar;
+
+    }
+    private static HBox textBar(){
+
+        //create menu at bottom
+        HBox bottombar=new HBox(10);
+        bottombar.setMinHeight(200);
+
+        TextArea textBox=new TextArea();
+        textBox.setWrapText(true);
+
+        //border for box
+        bottombar.setStyle("-fx-padding: 10;" +
+                "-fx-border-style: solid inside;" +
+                "-fx-border-width: 4;"  +
+                "-fx-border-radius: 5;" +
+                "-fx-border-color: #336699;");
+
+        //Button to send messages.
+        Button send = new Button("Send");
+        send.setPrefSize(100, 20);
+        send.setOnAction(new EventHandler<ActionEvent>() {
+            @Override
+            public void handle(ActionEvent actionEvent) {
+
+                //if contains bad word, alert box
```

Comment:
haha i like this.  you might want to connect with @rshar1, perhaps instead of bad word detection you show an alert box if sentiment is *really really* negative for a single message.